### PR TITLE
Add a modal component

### PR DESCRIPTION
**Pull request by mxmou**
### Changes

Moves the implementation of the More Settings modal into a separate component. This doesn't change anything, but it allows other modals to be added, specifically for preset switching.

### CODE REVIEW

File: webpages/settings/components/modal.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      },
+    },
+    methods: {
+      msg(...params) {
+        return this.$root.msg(...params);
+      },
+    },
+    events: {
+      clickOutside(e) {
+        if (this.isOpen && this.canCloseOutside && e.isTrusted) {
```
Comment (apple502j) [inline webpages/settings/components/modal.js:28]: Do we need isTrusted check?

Comment (WorldLanguages) [reply webpages/settings/components/modal.js:28]: This is probably code from stackoverflow?

Comment (mxmou) [reply webpages/settings/components/modal.js:28]: It was added in #1136 to ignore the click event from the hidden file input.

Comment (apple502j) [reply webpages/settings/components/modal.js:28]: ah, makes sense

**Pull request merged**